addons/mrp/models/mrp_workorder.py

```diff
@@ -304,23 +304,32 @@ def _float_duration_to_second(duration):
             return minutes * 60 + seconds
 
         for order in self:
-            old_order_duation = sum(order.time_ids.mapped('duration'))
+            old_order_duration = sum(order.time_ids.mapped('duration'))
             new_order_duration = order.duration
-            if new_order_duration == old_order_duation:
+            if new_order_duration == old_order_duration:
                 continue
 
-            delta_duration = new_order_duration - old_order_duation
+            delta_duration = new_order_duration - old_order_duration
 
             if delta_duration > 0:
-                date_start = datetime.now() - timedelta(seconds=_float_duration_to_second(delta_duration))
-                self.env['mrp.workcenter.productivity'].create(
-                    order._prepare_timeline_vals(delta_duration, date_start, datetime.now())
-                )
+                enddate = datetime.now()
+                date_start = enddate - timedelta(seconds=_float_duration_to_second(delta_duration))
+                if order.duration_expected >= new_order_duration or old_order_duration >= order.duration_expected:
+                    # either only productive or only performance (i.e. reduced speed) time respectively
+                    self.env['mrp.workcenter.productivity'].create(
+                        order._prepare_timeline_vals(new_order_duration, date_start, enddate)
+                    )
+                else:
+                    # split between productive and performance (i.e. reduced speed) times
+                    maxdate = fields.Datetime.from_string(enddate) - relativedelta(minutes=new_order_duration - order.duration_expected)
```

Comment:
It's not a better idea to do the split in _prepare_timeline_vals?

Reply:
If we do that then we would have to re-calculate the `old_order_duration` again in `_prepare_timeline_vals` since we need to know when to create 2 new split time_ids or only 1 "reduced speed" time_id based on what the duration of the already existing time_ids. It doesn't seem to add much value (at least for now) since the `_prepare_timeline_vals` isn't used anywhere else (except start button which will never need to start with a split time_id).